### PR TITLE
Make azure-pipelines-tasks-azurermdeploycommon compatible with Centauri

### PR DESCRIPTION
**Pull request by patelchandni**
**Task name**: Common/AzureRmDeploy-common

**Description**: 
1. Function apps on Centauri return 202 response (not 200) for PATCH /config/web.
2. Skip app restart steps with restart parameter is false.

**Documentation changes required:** (Y/N) NO

**Added unit tests:** (Y/N) NO

**Attached related issue:** (Y/N) NO

**Checklist**:
- [ ] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

**Comment by patelchandni** [comment]
AzureFunctionAppContainerV1 task is using the following:
1.  import { AzureAppService } from 'azure-pipelines-tasks-azurermdeploycommon/azure-arm-rest/azure-arm-app-service';
2. import { ContainerBasedDeploymentUtility } from 'azure-pipelines-tasks-azurermdeploycommon/operations/ContainerBasedDeploymentUtility';

Both need an update for Centauri so I have added these changes. Please let me know if this is the right location to reflect changes in the mentioned packages.

**Comment by patelchandni** [comment]
Also, Where can I bump the version?

File: Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export class ContainerBasedDeploymentUtility {
         this._appServiceUtility = new AzureAppServiceUtility(appService);
     }
 
-    public async deployWebAppImage(properties: any): Promise<void> {
+    public async deployWebAppImage(properties: any, restart:boolean=true): Promise<void> {
```
Comment (onetocny) [inline Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts:26]: nit: missing spaces

File: Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts [hunk 2]
```diff
@@ -46,8 +46,10 @@ export class ContainerBasedDeploymentUtility {
         tl.debug("Updating the webapp configuration.");
         await this._updateConfigurationDetails(properties["ConfigurationSettings"], properties["StartupCommand"], isLinuxApp, imageName, isMultiContainer, updatedMulticontainerConfigFile);
 
-        tl.debug('making a restart request to app service');
-        await this._appService.restart();
+        if (restart){
```
Comment (onetocny) [inline Tasks/Common/AzureRmDeploy-common/operations/ContainerBasedDeploymentUtility.ts:49]: nit: missing space

**Comment by onetocny** [comment]
As I mentioned in Team conversation these changes should be probably done in [this repository](https://github.com/microsoft/azure-pipelines-tasks-common-packages).

**Review by ozanovus — CHANGES_REQUESTED** [review]
New PR is created for actual library:
https://github.com/microsoft/azure-pipelines-tasks-common-packages/pull/143

We can abandon this one.

**Pull request closed without merging**